components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1836,73 +1836,93 @@ public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDoma
 
-                                    SAML2SSOMetadataConverter = metadataConverter;
+                                if (metadataConverter.canHandle(property)) {
 
                                     try {
-
-                                        metadata.append(properties[j].getValue());
-                                        StringBuilder certificate = new StringBuilder("");
+                                        if (isMetadataFileExist(identityProvider.getIdentityProviderName(),
+                                                property.getValue())) {
+                                            try {
+                                                metadataConverter.saveMetadataString(tenantId,
+                                                        identityProvider.getIdentityProviderName(),
+                                                        federatedAuthenticatorConfig.getName(), property.getValue());
+                                            } catch (IdentityProviderManagementException e) {
+                                                String data =
+                                                        "Couldn't save metadata in registry.SAML2SSOMetadataConverter is not set.";
+                                                throw IdPManagementUtil.handleServerException(
+                                                        IdPManagementConstants.ErrorMessage.ERROR_CODE_ADD_IDP, data);
+                                            }
+                                        }
+                                        StringBuilder certificate = new StringBuilder();
                                         try {
-                                            FederatedAuthenticatorConfig metaFederated = metadataConverter.getFederatedAuthenticatorConfig(properties, certificate);
+                                            FederatedAuthenticatorConfig metaFederated = metadataConverter
+                                                    .getFederatedAuthenticatorConfig(properties, certificate);
 
                                             String spName = "";
 
-                                            for (int b = 0; b < properties.length; b++) {
-                                                if (properties[b] != null && properties[b].getName() != null &&
-                                                        properties[b].getName().toString().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
-                                                    spName = properties[b].getValue();
+                                            for (Property value : properties) {
+                                                if (value != null && value.getName() != null &&
+                                                        value.getName()
+                                                                .equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
```

Comment:
Shall we invert the equals method to have the constant first so that we can omit the null check? WDYT?